src/cpp/rtps/security/SecurityManager.cpp

```diff
@@ -1167,7 +1165,6 @@ void SecurityManager::delete_participant_volatile_message_secure_pool()
         PoolConfig pool_cfg = PoolConfig::from_history_attributes(participant_volatile_message_secure_hattr_);
         participant_volatile_message_secure_pool_->release_history(pool_cfg, true);
         participant_volatile_message_secure_pool_->release_history(pool_cfg, false);
-        TopicPayloadPoolRegistry::release(participant_volatile_message_secure_pool_);
```

Comment:
Shouldn't we do a `participant_volatile_message_secure_pool_.reset()` here?

Reply:
certainly! I'm gonna check all former 'TopicPayloadPoolRegistry::release' calls again just in case 😅